app/scripts/transaction-manager.js

```diff
@@ -0,0 +1,207 @@
+const EventEmitter = require('events')
+const extend = require('xtend')
+const TxProviderUtil = require('./lib/provider-utils')
+
+module.exports = class TransactionManager extends EventEmitter {
+  constructor (opts) {
+    super()
+    this.txList = opts.TxListFromStore || []
+    this._persistTxList = opts.setTxList
+    this._unconfTxCbs = {}
+    this.txLimit = opts.txLimit
+    this.provider = opts.provider
+  }
+
+//   Returns the tx list
+  getTxList () {
+    return this.txList
+  }
+
+  // Saves the new/updated txList.
+  // Function is intended only for internal use
+  _saveTxList (txList) {
+    this.txList = txList
+    this._persistTxList(txList)
+  }
+
+  // Adds a tx to the txlist
+  addTx (txData, onTxDoneCb) {
+    var txList = this.getTxList()
+    var txLimit = this.txLimit
+    if (txList.length > txLimit - 1) {
+      txList.shift()
+    }
+    txList.push(txData)
+    this._saveTxList(txList)
+    this.addOnTxDoneCb(txData.id, onTxDoneCb)
+    this.emit('unapproved', txData)
+    this.emit('update')
+  }
+
+  // gets tx by Id and returns it
+  getTx (txId, cb) {
+    var txList = this.getTxList()
+    var tx = txList.find((tx) => tx.id === txId)
+    return cb ? cb(tx) : tx
+  }
+
+  //
+  updateTx (txData) {
+    var txId = txData.id
+    var txList = this.getTxList()
+
+    var updatedTxList = txList.map((tx) => {
+      if (tx.id === txId) {
+        tx = txData
+      }
+      return tx
+    })
+    this._saveTxList(updatedTxList)
+  }
+
+  get unConftxCount () {
+    return Object.keys(this.getUnapprovedTxList()).length
+  }
+
+  get pendingTxCount () {
+    return this.getTxsByMetaData('status', 'signed').length
+  }
+
+  getUnapprovedTxList () {
+    var txList = this.getTxList()
+    return txList.filter((tx) => {
+      return tx.status === 'unapproved'
+    }).reduce((result, tx) => {
+      result[tx.id] = tx
+      return result
+    }, {})
+  }
+
+  /*
+  Takes an object of fields to search for eg:
+  var thingsToLookFor = {
+    to: '0x0..',
+    from: '0x0..',
+    status: 'signed',
+  }
+  and returns a list of tx with all
+  options matching
+
+  this is for things like filtering a the tx list
+  for only tx's from 1 account
+  or for filltering for all txs from one account
+  and that have been 'confirmed'
+  */
+  getFilterdTxList (opts) {
+    var filteredTxList
+    Object.keys(opts).forEach((key) => {
+      filteredTxList = this.getTxsByMetaData(key, opts[key], filteredTxList)
```

Comment:
You could probably accomplish this with a single loop, calling `.filter()`, although it's kindof a micro-optimization.

The `filter` call would need to then return `false` if *any* of the provided opts didn't match, so in this case the loop would be inside the `filter()` call, avoiding storing intermediary lists.

This would also avoid needing to call `getTxsByMetaData`, which might improve readability?

Might then benefit from a different convenience method, called something like "doesTxMatchFilter".

Anyways, hyper-optimization note, non blocking.